File: 백준/Silver/7568. 덩치/덩치.kt
fun main() {

    val count = readln().toInt()

    val arr = Array(count) { IntArray(2) }
    var answer = ""

    for (i in 0 until count) {
        val (width, height) = readln().split(" ").map { it.toInt() }
        arr[i][0] = width
        arr[i][1] = height
    }

    for (i in 0 until count) {

        var rank = 1

        for (j in 0 until count) {
            if (i == j) continue
            if (arr[i][0] < arr[j][0] && arr[i][1] < arr[j][1]) {
                rank++
            }
        }
        answer = answer.plus(rank)
            .plus(" ")
    }
    println(answer.trim())

}

File: 백준/Silver/7568. 덩치/덩치.py
n = int(input())
answer = []
temp = []
for i in range(n):
    w, h = map(int, input().split())
    temp.append([w, h])

for i in range(n):
    count = 0
    for j in range(n):
        if temp[i][0] < temp[j][0] and temp[i][1] < temp[j][1]:
            count += 1
    answer.append(count + 1)

for item in answer:
    print(item, end = " ")